Doc/extending/newtypes_tutorial.rst

```diff
@@ -425,6 +425,18 @@ don't we have to do this?
 * when decrementing a reference count in a :c:member:`~PyTypeObject.tp_dealloc`
   handler on a type which doesn't support cyclic garbage collection [#]_.
 
+There is a much simpler way to reassign to ``self->first`` though, and that is by using :c:func:`Py_XSETREF` and :c:func:`Py_NewRef`::
+
+   if (first) {
+       Py_XSETREF(self->first, Py_NewRef(first));
+   }
+
+:c:func:`Py_XSETREF` automatically handles the assignment to ``self->first`` and the decrementing of its reference count using :c:func:`Py_XDECREF` (in the correct way that was shown earlier). :c:func:`Py_NewRef` automatically handles the incrementing of the reference count for ``first``. We can also apply this to ``self->last`` and ``last``::
+
+   if (last) {
+       Py_XSETREF(self->last, Py_NewRef(last));
+   }
```

Comment:
Why not suggesting to use these functions at the first place (in the code above)? I'm not sure that this tutorial is the best place to explain how to handle reference counting in length.

Reply:
I think there should be a part for `Py_SETREF`/`Py_XSETREF` and `Py_NewRef`/`Py_XNewRef` in the [reference counting part earlier in the tutorial](https://docs.python.org/3/extending/extending.html#reference-counts) and `Py_SETREF`/`Py_XSETREF` in the [reference counting section](https://docs.python.org/3/c-api/refcounting.html) since it isn't documented anywhere else yet. That way, this tutorial could just use `Py_XSETREF` and `Py_NewRef` instead of the 4-line snippet with no further explanation needed. I'll probably do that later if you agree with it.